packages/core/lib/console.js

```diff
@@ -262,7 +262,20 @@ class Console extends EventEmitter {
           path.join(this.options.contracts_build_directory, file),
           "utf8"
         );
-        jsonBlobs.push(JSON.parse(body));
+        const json = JSON.parse(body);
+        // hack to prevent a warning about the console.sol contract which we
+        // don't load into the environment - we only want to do this when it is
+        // Truffle's console.sol, that is why we check the sources keys
```

Comment:
I don't think this is a Hack, it is well reasoned logic :) 

```suggestion
        // Filter out Truffle's console.log. We don't want users to interact with in the REPL.
        // User contracts named console.log will be imported, and a warning will be issued.
```